### ISSUE TITLE
Unit-tests: missing documentation or support for simulating input

### ISSUE THREAD

**Issue by tugend** [open-wc/open-wc#998]
I've written a super simple input component that prohibits the user from typing anything that matches some regexp. 

I would really like to write a unit test to verify it worked, but can't find any examples, sufficient documentation or figure out how to get the inner input element to fire a change event similar to how it would be fired in the wild. 

I'm really hoping for something like enzymejs for react where I can do like this; `input.simulate('change', { target: { value: 'Hello' } })`

```
<custom-input sanitize='/[^\d+ ]/g'></custom-input>

class CustomInput {
    ...

    format(value) {
        return value.replace(this.sanitize, '');
    }

    onInput({target}) {
        target.value = this.format(target.value);
    }
}
```

Of course I hope you'd update the documentation so it might be easier for the next developer with  the same problem, as well as me. If there is something I misunderstand, I'll really appriciate an explanation before you close this issue.

Thanks in advance. ^_^

**Comment by tugend** [comment]
So this works. ^_^

```
export function simulateInput(target, value) {
  target.value = value;
  target.dispatchEvent(new InputEvent("input"));
}
```

**Comment by LarsDenBakker** [comment]
This would be a good library https://github.com/PolymerElements/iron-test-helpers/blob/master/mock-interactions.js. Some things are outdated, or specific to polymer 0.5 or 1/2/3. We could create a modernized version of this in `@open-wc/testing-helpers`.

I would prefer to simulate a keypress instead of setting the value property manually, if possible.

**Comment by tugend** [comment]
Thanks. I'll look into that. ^_^

**Comment by daKmoR** [comment]
imho such an abstraction is only marginally useful nowadays... when we still had to "deal" with keyCodes yeah it made sense... but now I don't see how the syntax could be much improved

this is vanilla
```js
el.dispatchEvent(new KeyboardEvent('keydown', { key: 'PageUp' }));
```

how would you write it "shorter"?
```js
simulateKey(el, 'PageUp', { type: 'keydown' });
```

if so then I would rather us vanilla as no one would need to be taught our own `simulateKey`.

Here is an example where it is used often and I think the tests are quite readable and would not really benefit from a `simulateKey` https://github.com/ing-bank/lion/blob/master/packages/calendar/test/lion-calendar.test.js#L714

**Comment by LarsDenBakker** [comment]
Good point! I wasnt aware that actually works, that fills in the input field too?

**Comment by tugend** [comment]
Haha! I agree completely with out @daKmoR.
Turns out that it really can be as simple. 

It didn't work for me, because I wanted to simulate the change of the input element as nested in another web component and then test the behavior of the parent component that actually change the dom. That is, it seemed to easily break and difficult to make work when I slightly changed implementations but kept behavior.

What I needed to learn was that 
(a) events that propagate upwards through shadow boundaries replace the target of the event with the last passed shadow component. 

"_Bubbling events fired from within shadow DOM are retargeted so that, to any listener external to your component, they appear to come from your component itself._"
https://lit-element.polymer-project.org/guide/events

(b) I have to set bubbles and composed to true for the event to pass shadow boundaries
"_To make a custom event pass through shadow DOM boundaries, you must set both the composed and bubbles flags to true_"

So if I want to simulate the behavior of an input event in the 'wild' I had to set those values too.
Now my tests actually behave as expected regardless of the component nesting. =D
```
export function simulateInput(target, value) {
  target.value = value;
  target.dispatchEvent(new InputEvent("input", { bubbles: true, composed: true}));
}
```

Thanks for input! =)
